src/state.rs

```diff
@@ -322,11 +344,11 @@ impl State {
         let kde_decoration_state = KdeDecorationState::new::<Self>(&dh, Mode::Client);
         let xdg_decoration_state = XdgDecorationState::new::<Self>(&dh);
         let session_lock_manager_state =
-            SessionLockManagerState::new::<Self, _>(&dh, client_has_security_context);
+            SessionLockManagerState::new::<Self, _>(&dh, client_should_see_privileged_protocols);
         XWaylandKeyboardGrabState::new::<Self>(&dh);
         PointerConstraintsState::new::<Self>(&dh);
         PointerGesturesState::new::<Self>(&dh);
-        SecurityContextState::new::<Self, _>(&dh, client_has_security_context);
+        SecurityContextState::new::<Self, _>(&dh, client_should_see_privileged_protocols);
```

Comment:
This would prevent `flatpak` from using the security context protocol if `COSMIC_ENABLE_WAYLAND_SECURITY` is set, since it wouldn't be started with a privileged socket. Would it be better to use `client_has_no_security_context` here?

Though it doesn't make a difference until we start taking into account the security context of Flatpak apps.

Reply:
> Would it be better to use client_has_no_security_context here?

Right, that wasn't on purpose, changing it back.

> Though it doesn't make a difference until we start taking into account the security context of Flatpak apps.

Well it should already block all "privileged" protocols. Though once we enforce security there is indeed no difference here, but given there might be in the future this is still a good change.